tests/e2e/performance/csi_tests/test_auto_reclaim_space_multi_clones.py

```diff
@@ -0,0 +1,243 @@
+
+
+@performance
+@skipif_ocs_version("<4.14")
+class TestReclaimSpaceCronJobMultiClones(PASTest):
+    """
+    This test verifies automatic creation of Reclaim Space Cron Jobs for multiple clones of RBD PVCs in
+    openshift-* namespace. It also verifies that no Reclaim Space Cron Job is created automatically any of the
+    multiple clones of CephFS PVC in this namespace
+    """
+
+    def setup(self):
+        logger.info("Starting the test setup")
+        super(TestReclaimSpaceCronJobMultiClones, self).setup()
+
+        self.num_of_clones = 512
+
+        # Getting the total Storage capacity
+        ceph_capacity = int(self.ceph_cluster.get_ceph_capacity())
+        # Getting the free Storage capacity
+        ceph_free_capacity = int(self.ceph_cluster.get_ceph_free_capacity())
+        # Use 70% of the free storage capacity in the test
+        capacity_to_use = int(ceph_free_capacity * 0.7)
+
+        # since we do not want to use more then 65%, we add 35% to the needed
+        # capacity, and minimum PVC size is 1 GiB
+        need_capacity = int((self.num_of_clones + 2) * 1.35)
+        # Test will run only on system with enough capacity
+        if capacity_to_use < need_capacity:
+            err_msg = (
+                f"The system has only {ceph_capacity} GiB, "
+                f"Of which {ceph_free_capacity} GiB is free, "
+                f"we want to use  {capacity_to_use} GiB, "
+                f"and we need {need_capacity} GiB to run the test"
+            )
+            logger.error(err_msg)
+            raise StorageNotSufficientException(err_msg)
+
+        # Calculating the PVC size in GiB
+        self.pvc_size = int(capacity_to_use / (self.num_of_clones + 2))
+
+        logger.info(
+            f"Total capacity size is : {ceph_capacity} GiB, "
+            f"Free capacity size is : {ceph_free_capacity} GiB, "
+            f"Going to use {need_capacity} GiB, "
+            f"With {self.num_of_clones} clones to {self.pvc_size} GiB PVC. "
+        )
+
+        namespace = f"openshift-{uuid4().hex}"
+        self.namespace = namespace
+        result = run_oc_command(cmd=f"create namespace {self.namespace}")
+        assert ERRMSG not in result[0], (
+            f"Failed to create namespace with name {namespace}" f"got result: {result}"
+        )
+        logger.info(f"Namespace {namespace} created")
+
+        self.cloned_obj_list = []
+
+    @pytest.fixture(autouse=True)
+    def teardown(self, request):
+        """
+        Cleanup the test environment
+        """
+
+        def finalizer():
+            logger.info("Starting the test cleanup")
+
+            # Delete the created clones
+            try:
+                for clone in self.cloned_obj_list:
+                    performance_lib.run_oc_command(
+                        cmd=f"delete pvc {clone}", namespace=self.namespace
+                    )
+            except Exception:
+                logger.warning("Clones were not deleted")
```

Comment:
since we have warning there is a possibility of printing this message and still continue to execute further cleanup i.e. before deleting clones PVC will be attempted to delete which might cause stale entries in PV.

Reply:
Is there a option we loop back or re iterate to perform the deletion again? to make it successful and continue with further cleanup?

Reply:
@ramkiperiy 
I see your point.
However, this is the way teardown works in all our tests - at first we delete the dependent entities ( like clones) and at the end we delete the PVC , pool and the namespace. The order of deletion is always opposite to the order of creation. 
Since this point is relevant not only to this test but to all the Performance tests - I'd suggest to leave it meanwhile as is and at some point to have a brainstorming to find the best way to cope with this issue.